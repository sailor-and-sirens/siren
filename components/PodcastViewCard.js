import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput, Image, ScrollView, Platform, Alert, TouchableOpacity } from 'react-native';
import PodcastEpisodeList from './PodcastEpisodeList';
import { Ionicons } from '@expo/vector-icons';
import { connect } from 'react-redux';
import { actionCreators as podcastsActions } from '../actions/Podcasts';
import { headerActions } from '../actions/Header';
import { subscribePodcast, getEpisodeDiscovery } from '../helpers';
import Spinner from 'react-native-loading-spinner-overlay';

const mapStateToProps = (state) => ({
  podcast: state.podcasts.currentPodcast,
  episodes: state.podcasts.podcastEpisodes,
  token: state.main.token,
  visible: state.podcasts.episodesLoading
});

class PodcastViewCard extends Component {

  render() {
    return (
      <View style={styles.wrapper}>
        <View style={styles.podcastView}>
          <View style={styles.leftView}>
            <Image source={{uri: this.props.podcast.artworkUrl600}} style={styles.image}/>
          </View>
          <View style={styles.rightView}>
            <Text style={styles.title} numberOfLines={1}>{this.props.podcast.collectionName}</Text>
            <Text style={styles.artist} numberOfLines={1}>{this.props.podcast.artistName}</Text>
            <ScrollView style={styles.descriptionView}>
              <Text style={styles.description}>{this.props.episodes[0].podcast.description.long}</Text>
            </ScrollView>
            <View style={styles.tagAddView}>
              <Text style={styles.tag} numberOfLines={1}> {this.props.podcast.primaryGenreName} </Text>
              <TouchableOpacity onPress={ () => {subscribePodcast(this.props); Alert.alert('Subscribed to ' + this.props.podcast.collectionName)}}>
                <Ionicons style={styles.subscribeIcon} size={25} color='grey' name="ios-add-circle-outline" />
              </TouchableOpacity>
            </View>
          </View>
        </View>
        <View style={styles.podcastBottomView}>
          <View style={styles.discoveryBar}>
            <TouchableOpacity onPress={ () => {this.getEpisodeDiscovery()}}>
              <Text style={styles.discovery} >Discover more podcasts like this</Text>
            </TouchableOpacity>
          </View>
        </View>
        <ScrollView style={styles.episodeScroll}>
        {this.props.visible ?
          <Spinner visible={this.props.visible} textContent={"Loading Episodes..."} textStyle={{color: '#FFF'}} />  :
          <PodcastEpisodeList/>}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
  },
  podcastView: {
    height: 160,
    flexDirection: 'row',
    paddingTop: 5,
    paddingBottom: 5,
    borderTopWidth: 2,
    borderTopColor: 'lightgrey',
  },
  leftView: {
    height: 150,
    width: 155,
    paddingLeft: 5
  },
  rightView: {
    paddingLeft: 3,
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'stretch',
    height: 150,
    paddingTop: 5,
    paddingRight: 2
  },
  podcastBottomView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 5,
    borderBottomWidth: 2,
    borderBottomColor: 'lightgrey'
  },
  episodeScroll: {
    marginBottom: 80,
  },
  descriptionView: {
    marginBottom: 5,
    height: 100,
  },
  image: {
    height: 150,
    width: 150,
  },
  description: {
     ...Platform.select({
      ios: {
        fontSize: 12,
      },
      android: {
        fontSize: 14,
      },
    }),
  },
  title: {
    fontWeight: "500",
    ...Platform.select({
      ios: {
        fontSize: 16,
      },
      android: {
        fontSize: 18,
      },
    }),
  },
  artist: {
  fontWeight: "400",
  ...Platform.select({
    ios: {
      fontSize: 13,
    },
    android: {
      fontSize: 14,
    },
  }),
  marginBottom: 5,
  },
  podcast: {
    fontWeight: "600",
    ...Platform.select({
      ios: {
        fontSize: 14,
      },
      android: {
        fontSize: 16,
      },
    }),
  },
  tagAddView: {
    flexDirection: 'row'
  },
  tag: {
    flex: 0.8,
    alignSelf: 'flex-end',
    marginRight: 5,
    padding: 2,
    backgroundColor: '#50BFB9',
    textAlign: 'center',
  },
  subscribeIcon: {
    flex: 0.2,
    marginBottom: -4,
    paddingRight: 5,
    textAlign: 'right'
  },
  discoveryBar: {
    flex: 1,
    paddingTop: 5,
    paddingBottom: 5,
  },
  discovery: {
    justifyContent: 'center',
    textAlign: 'center',
    fontSize: 14,
    fontWeight: '500',
  }
});

export default connect(mapStateToProps)(PodcastViewCard);
