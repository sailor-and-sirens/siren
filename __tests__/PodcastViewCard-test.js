import React from 'react';
import PodcastViewCard from '../components/PodcastViewCard.js';
import * as renderer from "react-test-renderer";
import { Provider } from 'react-redux';

const fakeStore = (state) => {
  return {
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {
      return { ...state }
    }
  };
}

const initialStateSwipe = {
  currentlyOpenSwipeable: false
}

const initialState = {
  episodesLoading: false,
  currentPodcast: {"wrapperType":"track", "kind":"podcast", "artistId":127981066, "collectionId":561470997, "trackId":561470997, "artistName":"WNYC Studios", "collectionName":"Note to Self", "trackName":"Note to Self", "collectionCensoredName":"Note to Self", "trackCensoredName":"Note to Self", "artistViewUrl":"https://itunes.apple.com/us/artist/wnyc/id127981066?mt=2&uo=4", "collectionViewUrl":"https://itunes.apple.com/us/podcast/note-to-self/id561470997?mt=2&uo=4", "feedUrl":"http://feeds.wnyc.org/notetoself-podcast", "trackViewUrl":"https://itunes.apple.com/us/podcast/note-to-self/id561470997?mt=2&uo=4", "artworkUrl30":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/30x30bb.jpg", "artworkUrl60":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/60x60bb.jpg", "artworkUrl100":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/100x100bb.jpg", "collectionPrice":0.00, "trackPrice":0.00, "trackRentalPrice":0, "collectionHdPrice":0, "trackHdPrice":0, "trackHdRentalPrice":0, "releaseDate":"2017-04-05T04:00:00Z", "collectionExplicitness":"cleaned", "trackExplicitness":"cleaned", "trackCount":87, "country":"USA", "currency":"USD", "primaryGenreName":"Tech News", "contentAdvisoryRating":"Clean", "artworkUrl600":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/600x600bb.jpg", "genreIds":["1448", "26", "1318"], "genres":["Tech News", "Podcasts", "Technology"]},
  podcastEpisodes: [
  {
    "title": "CARTA: Awareness of Death and Personal Mortality: Implications for Anthropogeny – Mind Over Reality Transition: The Evolution of Human Mortality Denial; Human Mortality Denial and Terror Management Theory; The Lure of Death: Suicide as a Uniquely Human Phenomenon",
    "podcast":{
      "description": {
        "long": "test"
      }
     },
    "enclosure": {
      "filesize": 27844374,
      "type": "audio/mpeg",
      "url": "http://podcast.uctv.tv/mp3/32048.mp3",
    },
    "guid": "32048carta-awareness-of-death-and-personal-mortality-implications-for-anthropogeny-–-“mind-over-reality-transition”-the-evolution-of-human-mortality-denial-human-mortality-denial-and-terror-management-theory-the-lure-of-death-suicide-as-a-uniquely-human-phenomenon",
    "published": "2017-05-24T21:00:00.000Z",
    "duration": "00:57:50",
    "description": "Ajit Varki explores the human capacity for denial of reality and how that has shaped our evolution; Sheldon Solomon different philosophies surrounding mortality; and Nicholas Humphrey provides a comprehensive look at the motivations for, prevalence of and reactions to the uniquely human act for suicide.  Series: &quot;CARTA - Center for Academic Research and Training in Anthropogeny&quot; [Humanities] [Science] [Show ID: 32048]"
  }]
}

const initialStateMain = {
  token: null,
  inboxFilters: null
}

const initialStatePlayer = {
  currentEpisode: null
}

const store = fakeStore({podcasts: initialState, main: initialStateMain, swipe: initialStateSwipe, player: initialStatePlayer});

test("Podcast view renders a podcast's info", () => {
  const tree = renderer.create(
    //wrap component in provide so it has access to store
    <Provider store={store}>
      <PodcastViewCard />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//To retake snapshot: npm test -- -u PodcastViewCard