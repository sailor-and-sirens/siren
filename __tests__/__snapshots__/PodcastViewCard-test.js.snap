// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Podcast view renders a podcast's info 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "lightgrey",
        "borderBottomWidth": 2,
        "borderTopColor": "lightgrey",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 155,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/600x600bb.jpg",
          }
        }
        style={
          Object {
            "height": 146,
            "width": 146,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 0.5,
          "height": 150,
          "justifyContent": "space-between",
          "paddingBottom": 3,
          "paddingLeft": 2,
          "paddingRight": 2,
          "paddingTop": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Note to Self
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 13,
            "fontWeight": "400",
            "marginBottom": 5,
          }
        }
      >
        WNYC Studios
      </Text>
      <ScrollView
        style={
          Object {
            "height": 100,
            "marginBottom": 4,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
            }
          }
        >
          test
        </Text>
      </ScrollView>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 1,
            "paddingRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#f4a442",
              "marginTop": 6,
              "padding": 2,
              "textAlign": "center",
              "width": 120,
            }
          }
        >
           
          Tech News
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "600",
              "height": 20,
              "justifyContent": "center",
              "textAlign": "center",
            }
          }
        >
          See more like this
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 250,
      }
    }
  >
    <ScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            leftActionActivationDistance={200}
            leftActionReleaseAnimationConfig={null}
            leftActionReleaseAnimationFn={null}
            leftButtonWidth={75}
            leftButtonsActivationDistance={75}
            leftButtonsCloseReleaseAnimationConfig={null}
            leftButtonsCloseReleaseAnimationFn={null}
            leftButtonsOpenReleaseAnimationConfig={null}
            leftButtonsOpenReleaseAnimationFn={null}
            onLayout={[Function]}
            onLeftActionActivate={[Function]}
            onLeftActionComplete={[Function]}
            onLeftActionDeactivate={[Function]}
            onLeftActionRelease={[Function]}
            onLeftButtonsActivate={[Function]}
            onLeftButtonsCloseComplete={[Function]}
            onLeftButtonsCloseRelease={[Function]}
            onLeftButtonsDeactivate={[Function]}
            onLeftButtonsOpenComplete={[Function]}
            onLeftButtonsOpenRelease={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onPanAnimatedValueRef={[Function]}
            onRef={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRightActionActivate={[Function]}
            onRightActionComplete={[Function]}
            onRightActionDeactivate={[Function]}
            onRightActionRelease={[Function]}
            onRightButtonsActivate={[Function]}
            onRightButtonsCloseComplete={[Function]}
            onRightButtonsCloseRelease={[Function]}
            onRightButtonsDeactivate={[Function]}
            onRightButtonsOpenComplete={[Function]}
            onRightButtonsOpenRelease={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onSwipeComplete={[Function]}
            onSwipeMove={[Function]}
            onSwipeRelease={[Function]}
            onSwipeStart={[Function]}
            rightActionActivationDistance={125}
            rightActionReleaseAnimationConfig={null}
            rightActionReleaseAnimationFn={null}
            rightButtonWidth={75}
            rightButtonsActivationDistance={75}
            rightButtonsCloseReleaseAnimationConfig={null}
            rightButtonsCloseReleaseAnimationFn={null}
            rightButtonsOpenReleaseAnimationConfig={null}
            rightButtonsOpenReleaseAnimationFn={null}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
            swipeReleaseAnimationConfig={
              Object {
                "duration": 250,
                "easing": [Function],
                "toValue": Object {
                  "x": 0,
                  "y": 0,
                },
              }
            }
            swipeReleaseAnimationFn={[Function]}
            swipeStartMinDistance={15}
          >
            <View
              style={
                Object {
                  "marginLeft": -0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 0,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "justifyContent": "center",
                      "paddingRight": 20,
                    },
                    Object {
                      "backgroundColor": "#42f4c5",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Add to Inbox
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgrey",
                    "borderBottomWidth": 2,
                    "borderTopColor": "lightgrey",
                    "height": 100,
                    "justifyContent": "space-between",
                    "marginLeft": 6,
                    "marginRight": 2,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.75,
                      "flexDirection": "row",
                      "height": 80,
                      "justifyContent": "space-between",
                      "marginBottom": 4,
                      "marginTop": 4,
                      "paddingRight": 5,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "flex": 0.75,
                        "height": 80,
                        "justifyContent": "space-around",
                        "paddingLeft": 10,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "fontSize": 14,
                          "fontWeight": "500",
                          "marginBottom": 5,
                        }
                      }
                    >
                      CARTA: Awareness of Death and Personal Mortality: Implications for Anthropogeny – Mind Over Reality Transition: The Evolution of Human Mortality Denial; Human Mortality Denial and Terror Management Theory; The Lure of Death: Suicide as a Uniquely Human Phenomenon
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={2}
                      style={
                        Object {
                          "fontSize": 12,
                          "fontWeight": "400",
                        }
                      }
                    >
                      Ajit Varki explores the human capacity for denial of reality and how that has shaped our evolution; Sheldon Solomon different philosophies surrounding mortality; and Nicholas Humphrey provides a comprehensive look at the motivations for, prevalence of and reactions to the uniquely human act for suicide.  Series: &quot;CARTA - Center for Academic Research and Training in Anthropogeny&quot; [Humanities] [Science] [Show ID: 32048]
                    </Text>
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "marginBottom": 5,
                          "marginTop": 8,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "grey",
                            "fontSize": 13,
                            "fontWeight": "400",
                          }
                        }
                      >
                        Wed, 24 May 2017
                      </Text>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "marginBottom": 3,
                            "marginRight": 10,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 15,
                              "marginRight": 7,
                              "width": 15,
                            }
                          }
                        />
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "grey",
                              "fontSize": 13,
                              "fontWeight": "400",
                              "marginRight": 5,
                            }
                          }
                        >
                          00:57:50
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </ScrollView>
  </View>
</View>
`;
